interpreter/value_deployedcontract.go

```diff
@@ -94,18 +94,18 @@ func NewNativeDeployedContractPublicTypesFunctionValue(
 			)
 			name = AssertValueOfType[*StringValue](nameFieldValue)
 		} else {
+			// Interpreter provides address and name
 			address = *addressPointer
+			if name == nil {
+				panic(errors.NewUnreachableError())
+			}
 		}
 
-		if publicTypes == nil {
```

Comment:
The result used to be cached for the interpreter, but also accidentally for the VM. As we don't cache in any other places, I thought it was better/cleaner/safer to remove the caching here and always compute the result, no matter the environment.